djangocms_fil_bootstrap/components/collections.py

```diff
@@ -9,51 +9,25 @@ class Collections(Component):
     default_factory = dict
 
     def parse(self):
-        if not self.raw_data.get("enable", False):
-            return
-        moderator = self.bootstrap.users["moderator"]
-        moderator2 = self.bootstrap.users["moderator2"]
-        moderator3 = self.bootstrap.users["moderator3"]
-        collection1 = ModerationCollection.objects.create(
-            author=moderator,
-            name="Collection 1",
-            workflow=self.bootstrap.workflows["wf4"],
-        )
-        collection1.add_version(get_version(self.bootstrap.pages["page1"]))
-        collection1.add_version(get_version(self.bootstrap.pages["page2"]))
-
-        collection2 = ModerationCollection.objects.create(
-            author=moderator2,
-            name="Collection 2",
-            workflow=self.bootstrap.workflows["wf4"],
-        )
-        collection2.add_version(get_version(self.bootstrap.pages["page3"]))
-        collection2.add_version(get_version(self.bootstrap.pages["page4"]))
-
-        collection3 = ModerationCollection.objects.create(
-            author=moderator3,
-            name="Collection 3",
-            workflow=self.bootstrap.workflows["wf4"],
-        )
-        collection3.add_version(get_version(self.bootstrap.pages["page5"]))
-
-        collection4 = ModerationCollection.objects.create(
-            author=moderator,
-            name="Collection 4",
-            workflow=self.bootstrap.workflows["wf4"],
-        )
-        collection4.add_version(get_version(self.bootstrap.pages["page6"]))
-
-        collection5 = ModerationCollection.objects.create(
-            author=moderator2,
-            name="Collection 5",
-            workflow=self.bootstrap.workflows["wf4"],
-        )
-        collection5.add_version(get_version(self.bootstrap.pages["page7"]))
-
-        collection6 = ModerationCollection.objects.create(
-            author=moderator3,
-            name="Collection 6",
-            workflow=self.bootstrap.workflows["wf5"],
-        )
-        collection6.add_version(get_version(self.bootstrap.pages["page8"]))
+        for name, data in self.raw_data.items():
+            collection_data = {
+                "author": self.bootstrap.users[data["user"]],
+                "workflow": self.bootstrap.workflows[data["workflow"]],
+                "name": data["name"],
+            }
+            collection = self.get_or_create(collection_data)
+            for page in data.get("pages", []):
+                # access pages from bootstrap as if a dict because the __getitem__ method in Bootstrap
+                # allows this and without it mock objects will fail via other access methods.
+                self.add_version(collection, self.get_version(self.bootstrap.pages[page]))
+            self.data[name] = collection
+
+    def get_version(self, page):
```

Comment:
This method doesn't do anything? can we not just use get_version rather making this as a class method?

Reply:
If we do that, that then we can't mock get_version which is necessary for test_parse. Please pull and try the tests again, should run now. Tx.

Reply:
you are mocking get_version as a class method which I think you can do same for a method that you are importing from the helper/utils.